public/js/tracking_protection.js

```diff
@@ -44,5 +54,18 @@ if (!_dntEnabled()) {
 if(typeof(ga) !== "undefined") {
 	ga("create", "UA-77033033-16");
 	ga("set", "anonymizeIp", true);
-	ga("send", "pageview");
+
+
+  // Strip url params if they contain token or hash
+  const loc = document.location;
+  let pageValue = loc.pathname + loc.search;
+  if (document.location.search &&
+      (document.location.search.indexOf("token") !== -1 ||
```

Comment:
Sorry, I'm not 100% certain what we're trying to solve, but this feels almost too greedy.
What would happen if we had a breach Name like `/?breach=Bitcointokens`?

We don't seem to have any matches currently, but it could happen at any random time based on the breach Name that Troy gives a future breach.

```js
const got = require("got");

async function main(prod=false) {
  const breaches = await got(prod ? "https://haveibeenpwned.com/api/v2/breaches" : "https://stage.haveibeenpwned.com/api/v2/breaches", {json:true});

  breaches.body
    .filter(b => b.Name.includes("token") || b.Name.includes("hash"))
    .forEach(b => console.log(b.Name));
}

main();
```

Reply:
Trying to strip the url params off the `pageValue` sent to Google Analytics when the url params include either the user `token` or the user `hash`, which we want to prevent from "leaking" out as much as possible. (This also makes it easier to see the overall aggregate user flow instead of seeing tons of separate pageviews at individualized `/user/verify?token=...` and `/user/unsubscribe?token=...` urls)

I updated to look for `token=` or `hash=` in the url. And I also updated to set both the `location` and `page` properties of ga, and to trim the `Referer` header of calls to GA so they don't leak any of these values either.